community/src/community.cpp

```diff
@@ -2176,6 +2316,10 @@ bool community::is_amend_action(name calling_action) {
            calling_action == set_vote_rule_action;
 }
 
+uint128_t community::build_reference_id(uint64_t reference_id, uint64_t type) {
+    return static_cast<uint128_t>(type)  | static_cast<uint128_t>(reference_id) << 64;
+}
```

Comment:
- Use inline function and put in header file.
- better to modify as  below, because when you pull the list, you can see sequence badge ids,   sequence postion ids ...
` return static_cast<uint128_t>(reference_id)  | static_cast<uint128_t>(type) << 64;`